src/mrinufft/trajectories/tools.py

```diff
@@ -700,3 +701,178 @@ def radialize_center(trajectory, nb_samples, in_out=False):
         return _radialize_in_out(trajectory, nb_samples)
     else:
         return _radialize_center_out(trajectory, nb_samples)
+
+
+#################
+# Randomization #
+#################
+
+
+def _flip2center(mask_cols: list[int], center_value: int) -> np.ndarray:
+    """
+    Reorder a list by starting by a center_position and alternating left/right.
+
+    Parameters
+    ----------
+    mask_cols: list or np.array
+        List of columns to reorder.
+    center_pos: int
+        Position of the center column.
+
+    Returns
+    -------
+    np.array: reordered columns.
+    """
+    center_pos = np.argmin(np.abs(np.array(mask_cols) - center_value))
+    mask_cols = list(mask_cols)
+    left = mask_cols[center_pos::-1]
+    right = mask_cols[center_pos + 1 :]
+    new_cols = []
+    while left or right:
+        if left:
+            new_cols.append(left.pop(0))
+        if right:
+            new_cols.append(right.pop(0))
+    return np.array(new_cols)
+
+
+def get_random_loc_1d(
+    dim_size,
+    center_prop,
+    accel=4,
+    pdf="uniform",
+    rng=None,
+    order="center-out",
+) -> np.ndarray:
+    """Get slice index at a random position.
+
+    Parameters
+    ----------
+    dim_size: int
+        Dimension size
+    center_prop: float or int
+        Proportion of center of kspace to continuouly sample
+    accel: float
+        Undersampling/Acceleration factor
+    pdf: str, optional
+        Probability density function for the remaining samples.
+        "gaussian" (default) or "uniform".
```

Comment:
In `stack_random` below, this argument can also be an array. I don't think this function can handle it, does it ?

Reply:
it does.